src/NHibernate/Loader/Loader.cs

```diff
@@ -937,6 +947,17 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 					obj = session.GetEntityUsingInterceptor(key);
 					var alreadyLoaded = obj != null;
 					var persister = persisters[i];
+					if (IsChildFetchEntity(i))
+					{
+						if (!alreadyLoaded && mustLoadMissingEntity)
+						{
+							// Missing in session while its data has not been selected: fallback on immediate load
+							obj = session.ImmediateLoad(key.EntityName, key.Identifier);
```

Comment:
I have first attempted to do an `InternalLoad` here instead, without accounting for `returnProxies`/`mustLoadMissingEntity`, and without changing anything to `GetRow`caller. But this was (as expected) potentially yielding a proxy, which `GetRow` caller was not supporting: it was forcing the proxy initialization with itself as implementation, resulting in stack overflows when attempting to access the entity state.

For this reason, when proxy are supported, better yield `null` here and let `GetRow` caller handle returning a proxy.